File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        obj = {}
        for n in nums:
            obj[n] = obj.get(n, 0) + 1
        
        lst = sorted(list(map(lambda x: [obj[x], x], obj.keys())), reverse=True)

        res = []
        
        for i in range(k):
            res.append(lst[i][1])
        
        return res

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {

    const obj = {};
    nums.forEach(n => obj[n] = (obj[n] || 0) + 1);

    const list = Object.keys(obj).map(k => [obj[k], k]);
    list.sort((a, b) => b[0] - a[0]);

    const res = [];

    for (let i = 0; i < k; i++) {
        res.push(Number(list[i][1]));
    }

    return res;
    
};